design-book/src/vision.md

```diff
@@ -0,0 +1,26 @@
+# Vision
+
+**TL;DR:** Think Valve's Hammer editor not Unity-but-Rust.
+
+Bevy's fabled editor has been the subject of much debate and frustration over the years,
+but to actually come together and build it, the team needs to align on a vision for its future.
+What is an editor? Why are we building it? What will change? Why do people think that this will push Bevy from a mere "framework" to a "real game engine"?
+
+Bevy's current strengths are simple but compelling:
+
+- delightful ECS-based architecture: fast, flexible and frankly pleasant
+- absurdly flexible: remove anything you want, and grab great new things from the ecosystem
+- Rust all the way down, making refactoring, building, testing, distribution and improving things upstream painless
+
+In the context of an "editor" though, the current "code-only" approach has several critical limitations that make it infeasible for all but the smallest and most technical of commercial gamedev teams:
+
+- artists (illustrators, animators, sound effects, music) and designers (gameplay, systems, level) are entirely reliant on more technical team members to integrate and test their work in the game
+- authoring game-specific content (custom materials, levels, designed content like items) requires custom tooling
+- Bevy developers are left to piece together their own testing and debugging workflow
+- Rust's compile times make some iteration-heavy tasks (like adjusting UI or tweaking lighting) painfully slow
```

Comment:
Many highly skilled programmers prefer not to spend time writing boilerplate heavy code to define scenes.

Reply:
Yeah, I can confirm, our team (which is 100% programmers) spent a lot of time trying to avoid having to write down a tonne of boilerplate to put together a level. (We ended up using a semi-automated tool that builds the level programmatically from geometrical and gameplay constraints, so the editor is not super critical. But if we did have an editor, we may have went a completely different direction.)